homeassistant/components/ping/sensor.py

```diff
@@ -0,0 +1,117 @@
+"""Sensor platform that for Ping integration."""
+from collections.abc import Callable
+from dataclasses import dataclass
+
+from homeassistant.components.sensor import (
+    SensorDeviceClass,
+    SensorEntity,
+    SensorEntityDescription,
+    SensorStateClass,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import EntityCategory, UnitOfTime
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import PingDomainData
+from .const import DOMAIN
+from .coordinator import PingResult, PingUpdateCoordinator
+from .entity import PingEntity
+
+
+@dataclass(frozen=True, kw_only=True)
+class PingSensorEntityDescription(SensorEntityDescription):
+    """Class to describe a Ping sensor entity."""
+
+    value_fn: Callable[[PingResult], float | None]
+    has_fn: Callable[[PingResult], bool]
```

Comment:
Does it not always have these?

Reply:
They are actually different. `mdev` is only there if the fallback ping subprocess is used, `icmplib` does not provide this.